src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandlePropertyName.cs

```diff
@@ -82,6 +82,27 @@ public static partial class JsonSerializer
                 }
                 else
                 {
+                    // Support JsonException.Path.
+                    Debug.Assert(
+                        state.Current.JsonPropertyInfo.JsonPropertyName == null ||
+                        options.PropertyNameCaseInsensitive ||
+                        propertyName.SequenceEqual(state.Current.JsonPropertyInfo.JsonPropertyName));
+
+                    if (state.Current.JsonPropertyInfo.JsonPropertyName == null)
+                    {
+                        if (options.PropertyNameCaseInsensitive)
+                        {
+                            // Each payload can have a different name here; remember the value on the temporary stack.
+                            state.Current.JsonPropertyName = propertyName.ToArray();
```

Comment:
Store `propertyName.ToArray()` in a local since its common between both branches.

Reply:
Done